### PR TITLE
Search on release page

### PR DESCRIPTION
**Pull request by hedgehog1833**
The release page has a search field and calls the butler with all existing filter taken into consideration.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Releases page: Search for artist name or album name](https://trello.com/c/N7R5Fa5l/228-releases-page-search-for-artist-name-or-album-name)

File: butler/src/main/java/rocks/metaldetector/butler/client/transformer/ButlerReleaseRequestTransformer.java [hunk 1]
```diff
@@ -8,13 +8,14 @@
...
-            .artists(artists)
-            .dateFrom(timeRange.getDateFrom())
-            .dateTo(timeRange.getDateTo())
-            .build();
+        .page(pageRequest != null ? pageRequest.getPage() : 0)
+        .size(pageRequest != null ? pageRequest.getSize() : 0)
+        .artists(artists)
+        .dateFrom(timeRange.getDateFrom())
+        .dateTo(timeRange.getDateTo())
+        .query(query == null || query.trim().isEmpty() ? null : query)
```
Comment (DanielW1987) [inline butler/src/main/java/rocks/metaldetector/butler/client/transformer/ButlerReleaseRequestTransformer.java:18]:
```suggestion
        .query(query == null || query.isBlank() ? null : query)
```

File: webapp/src/main/resources/static/ts/src/service/releases-service.ts [hunk 2]
```diff
@@ -116,6 +124,8 @@ export class ReleasesService {
         urlSearchParams.set(ReleasesService.RELEASES_PARAM_NAME, releasesFilterValue);
         urlSearchParams.set(ReleasesService.SORT_BY_PARAM_NAME, sortBy);
         urlSearchParams.set(ReleasesService.SORT_DIRECTION_PARAM_NAME, sortDirection);
+        urlSearchParams.set("query", this.searchField.value);
+
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/ts/src/service/releases-service.ts:128]: please use constant here

File: webapp/src/main/resources/static/ts/src/service/releases-service.ts [hunk 3]
```diff
@@ -47,12 +49,14 @@ export class ReleasesService {
...
         this.sortAscRb = document.getElementById("sort-asc-rb") as HTMLInputElement;
         this.sortDescRb = document.getElementById("sort-desc-rb") as HTMLInputElement;
+        this.searchField = document.getElementById("release-search") as HTMLInputElement;
     }
 
     private initFilterValuesFromUrl(): void {
         const releasesParamValue = this.urlService.getParameterFromUrl("releases");
         const sortParamValue = this.urlService.getParameterFromUrl("sort");
         const directionParamValue = this.urlService.getParameterFromUrl("direction");
+        const searchQueryParamValue = this.urlService.getParameterFromUrl("query");
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/ts/src/service/releases-service.ts:59]: please use constant here and refactor the statements above

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**